### PR TITLE
fix(native-filters): Fix indicators

### PR DESCRIPTION
**Pull request by simcha90**
### SUMMARY
Hide unfeatured filters in FiltersBade

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by amitmiran137** [comment]
resolves https://github.com/apache/superset/issues/14333

File: superset-frontend/src/dashboard/components/FiltersBadge/selectors.ts [hunk 1]
```diff
@@ -220,57 +221,62 @@ export const selectNativeIndicatorsForChart = (
...
-        column,
-        name: nativeFilter.name,
-        path: [nativeFilter.id],
-        status: getStatus({ value, isAffectedByScope, column }),
-        value,
-      };
-    },
-  );
+  let nativeFilterIndicators: any = [];
+  if (isFeatureEnabled(FeatureFlag.DASHBOARD_NATIVE_FILTERS)) {
```
Comment (amitmiran137) [inline superset-frontend/src/dashboard/components/FiltersBadge/selectors.ts:225]: I don't think filter indicator is intended only for native filters
@villebro WDYT?

Comment (villebro) [reply superset-frontend/src/dashboard/components/FiltersBadge/selectors.ts:225]: I think this is correct - filter box indicators are handled separately by `selectIndicatorsForChart`

File: superset-frontend/src/dashboard/components/FiltersBadge/selectors.ts [hunk 2]
```diff
@@ -220,57 +221,62 @@ export const selectNativeIndicatorsForChart = (
...
-      const scope = chartConfig?.crossFilters?.scope;
-      const isAffectedByScope = getTreeCheckedItems(
-        scope,
-        dashboardLayout,
-      ).some(
-        layoutItem => dashboardLayout[layoutItem]?.meta?.chartId === chartId,
-      );
+  let crossFilterIndicators: any = [];
+  if (isFeatureEnabled(FeatureFlag.DASHBOARD_CROSS_FILTERS)) {
+    crossFilterIndicators = Object.values(chartConfiguration).map(
```
Comment (villebro) [inline superset-frontend/src/dashboard/components/FiltersBadge/selectors.ts:252]: Right now it shows all charts that have cross filter behavior as potentially emitting charts, which is not the case unless they've been enabled. I'd almost vote for adding a `.filter(filter => filter.status === IndicatorStatus.CrossFilterApplied)` at the end of this `map` to remove any cross filters that aren't applied for now, as listing them among unset native filters seems confusing.

**Review by villebro — COMMENTED** [review]
Code LGTM, but I'd vote for removing unset cross filters from the unset filters all together as we'll probably want to enable cross filtering support in all charts sooner or later.

File: superset-frontend/src/dashboard/components/FiltersBadge/selectors.ts (hunk 2 above)
Comment (simcha90) [reply superset-frontend/src/dashboard/components/FiltersBadge/selectors.ts:252]: I think it's like in filters that not applied for current chart it shows them in `Unset` section, but it says that this Cross charts they can be applied to current chart, from point of current, I think... @amitmiran137

**Review by villebro — APPROVED** [review]

**Pull request merged**